### PR TITLE
[Docs] Guidance for choosing redirectUris

### PR DESCRIPTION
**Pull request by tnorling**
Adds docs explaining best practices for what your `redirectUri` should and should not do. Also updates an error message to point to docs with a better explanation.

### CODE REVIEW

File: lib/msal-browser/docs/errors.md [hunk 1]
```diff
@@ -120,6 +121,26 @@ If you are unable to figure out why this error is being thrown please [open an i
...
+
+This error is thrown when calling `ssoSilent` or `acquireTokenSilent` and the page used as your `redirectUri` is attempting to invoke a login or acquireToken function.
+Our recommended mitigation for this is to set your `redirectUri` to a blank page that does not implement MSAL when invoking silent APIs. This will also have the added benefit of improving performance as the hidden iframe doesn't need to render your page.
+
+✔️ You can do this on a per request basis, for example:
+
+```javascript
+msalInstance.acquireTokenSilent({
+    scopes: ["User.Read"],
+    redirectUri: "http://localhost:3000/blank"
```
Comment (jasonnutter) [inline lib/msal-browser/docs/errors.md:136]: I would suggest including the html extension to make it clear what this route should be:
```suggestion
    redirectUri: "http://localhost:3000/blank.html"
```

**Review by jasonnutter — APPROVED** [review]
One comment otherwise looks good!

**Review by jo-arroyo — APPROVED** [review]

File: lib/msal-browser/docs/initialization.md [hunk 2]
```diff
@@ -94,11 +94,23 @@ In the browser, there are two ways you can present the login screen to your user
 
 The popup APIs use ES6 Promises that resolve when the authentication flow in the popup concludes and returns to the redirect URI specified, or reject if there are issues in the code or the popup is blocked.
 
+#### RedirectUri Considerations
+
+When using popup APIs we recommend setting the `redirectUri` to a blank page or a page that does not implement MSAL. This will help prevent potential issues as well as improve performance. If your application is only using popup and silent APIs you can set this on the `PublicClientApplication` config. If your application also needs to support redirect APIs you can set the `redirectUri` on a per request basis:
+
+```javascript
+msalInstance.loginPopup({
+    redirectUri: "http://localhost:3000/blank"
```
Comment (tnorling) [inline lib/msal-browser/docs/initialization.md:103]:
```suggestion
    redirectUri: "http://localhost:3000/blank.html"
```

File: lib/msal-browser/docs/login-user.md [hunk 3]
```diff
@@ -124,6 +124,18 @@ try {
...
 
+## RedirectUri Considerations
+
+When using popup and silent APIs we recommend setting the `redirectUri` to a blank page or a page that does not implement MSAL. This will help prevent potential issues as well as improve performance. If your application is only using popup and silent APIs you can set this on the `PublicClientApplication` config. If your application also needs to support redirect APIs you can set the `redirectUri` on a per request basis:
+
+Note: This does not apply for `loginRedirect` or `acquireTokenRedirect`. When using those APIs please see the directions on handling redirects [here](./initialization#redirect-apis)
+
+```javascript
+msalInstance.loginPopup({
+    redirectUri: "http://localhost:3000/blank"
```
Comment (tnorling) [inline lib/msal-browser/docs/login-user.md:135]:
```suggestion
    redirectUri: "http://localhost:3000/blank.html"
```

**Pull request merged**